sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/bucket.scala

```diff
@@ -29,7 +32,8 @@ package org.apache.spark.sql.execution.datasources
 private[sql] case class BucketSpec(
     numBuckets: Int,
     bucketColumnNames: Seq[String],
-    sortColumnNames: Seq[String])
+    sortColumnNames: Seq[String],
+    sortDirections: Seq[SortDirection] = Nil)
```

Comment:
I still think sortColumnNames and sortDirections should be combined into a seq of pairs.